lib/Limiters/IpLimiter.php

```diff
@@ -19,25 +19,32 @@ class IpLimiter extends UserPassBaseLimiter
     public function __construct(Configuration $config)
     {
         parent::__construct($config);
+        /** @var string[] whitelist */
         $this->whitelist = $config->getArray('whitelist', []);
     }
 
     public function allow(string $username, string $password): string
     {
-        if ($this->isIpWhiteListed($_SERVER['REMOTE_ADDR'])) {
+        if ($this->isIpWhiteListed($this->getClientIp())) {
             return UserPassBaseLimiter::PREAUTH_CONTINUE;
         }
         return parent::allow($username, $password);
     }
 
     public function postFailure(string $username, string $password): int
     {
-        if ($this->isIpWhiteListed($_SERVER['REMOTE_ADDR'])) {
+        if ($this->isIpWhiteListed($this->getClientIp())) {
             return 0;
         }
         return parent::postFailure($username, $password);
     }
 
+    private function getClientIp(): string
+    {
+        /** @var string */
+        return $_SERVER['REMOTE_ADDR'];
```

Comment:
```
use Symfony\Component\HttpFoundation\Request;

$request = Request::createFromGlobals();
return $request->server->get('REMOTE_ADDR');
```

This should be more future proof.

Reply:
Thanks, I'll make that change in my next update.